### PR TITLE
feat(cloud): add new endpoint for resource access management listing

### PR DESCRIPTION
**Pull request by adr-mo**
## Description

🏷️ MON-24351

Add listing endpoint for the Resource Access Management page (cloud exclusive)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [x] 24.04.x (master)

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/www/install/php/Update-24.04.0.php [hunk 1]
```diff
@@ -58,6 +80,11 @@
     $errorMessage = 'Unable to insert topology for Resource Access Management';
     $insertTopologyForResourceAccessManagement($pearDB);
 
+    $errorMessage = 'Unable to add columns description and create_from_cloud to acl_groups table';
```
Comment (adr-mo) [inline centreon/www/install/php/Update-24.04.0.php:83]:
```suggestion
    $errorMessage = 'Unable to add columns cloud_description and cloud_specific to acl_groups table';
```

File: centreon/src/Core/ResourceAccess/Application/UseCase/FindRules/FindRules.php [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+            return;
+        }
+
+        try {
+            $rules = $this->repository->findAllByRequestParameters($this->requestParameters);
+            $presenter->presentResponse(
+                $this->createResponse($rules)
+            );
+        } catch (\Throwable $ex) {
```
Comment (adr-mo) [inline centreon/src/Core/ResourceAccess/Application/UseCase/FindRules/FindRules.php:77]: add missing catch for an exception raised by the requestParameters

File: centreon/src/Core/ResourceAccess/Application/UseCase/FindRules/FindRulesResponse.php [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+ */
+
+declare(strict_types = 1);
+
+namespace Core\ResourceAccess\Application\UseCase\FindRules;
+
+final class FindRulesResponse
+{
+    /** @var RuleDTO[] */
+    public array $rulesDTO = [];
```
Comment (adr-mo) [inline centreon/src/Core/ResourceAccess/Application/UseCase/FindRules/FindRulesResponse.php:29]: DTO to Dto

File: centreon/src/Core/ResourceAccess/Application/UseCase/FindRules/RuleDTO.php [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types = 1);
+
+namespace Core\ResourceAccess\Application\UseCase\FindRules;
+
+final class RuleDTO
```
Comment (adr-mo) [inline centreon/src/Core/ResourceAccess/Application/UseCase/FindRules/RuleDTO.php:26]: DTO => Dto

File: centreon/tests/php/Core/ResourceAccess/Application/UseCase/FindRules/FindRulesTest.php [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+        ]);
+
+    ($this->useCase)($this->presenter);
+    expect($this->presenter->response)
+        ->toBeInstanceOf(ForbiddenResponse::class)
+        ->and($this->presenter->response->getMessage())
+        ->toBe(RuleException::notAllowed()->getMessage());
+});
+
+it('should present an ErrorResponse when an exception occurs', function (): void {
```
Comment (adr-mo) [inline centreon/tests/php/Core/ResourceAccess/Application/UseCase/FindRules/FindRulesTest.php:63]: add test for request parameter exception

File: centreon/tests/php/Core/ResourceAccess/Domain/Model/RuleTest.php [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+
+declare(strict_types=1);
+
+namespace Tests\Core\ResourceAccess\Domain\Model;
+
+use Assert\InvalidArgumentException;
+use Centreon\Domain\Common\Assertion\AssertionException;
+use Core\ResourceAccess\Domain\Model\Rule;
+
+beforeEach(function (): void {
```
Comment (adr-mo) [inline centreon/tests/php/Core/ResourceAccess/Domain/Model/RuleTest.php:30]: Test postive int for id

**Review by jeremyjaouen — APPROVED** [review]

**Review by dmyios — APPROVED** [review]

**Review by a-launois — APPROVED** [review]

File: centreon/src/Core/ResourceAccess/Domain/Model/Rule.php [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+    {
+        return $this->name;
+    }
+
+    /**
+     * @param null|string $description
+     *
+     * @return self
+     */
+    public function setDescription(?string $description): self
```
Comment (callapa) [inline centreon/src/Core/ResourceAccess/Domain/Model/Rule.php:75]: Not valid if we consider the object to be immutable, but this subject is under discussion so ok.

**Review by callapa — APPROVED** [review]

**Review by wtermellil — APPROVED** [review]

**Pull request merged**